src/gwt/src/org/rstudio/core/client/VirtualConsole.java

```diff
@@ -695,14 +732,42 @@ public void ensureStartingOnNewLine()
             submit("\n");
       }
    }
+   private class Hyperlink
+   {
+      public Hyperlink(String url, String params) {
+         url_ = url;
+         params_ = params;
+      }
+
+      public String getTitle() {
+         return url_;
+      }
+
+      public String url_;
+      public String params_;
+   }
    private class ClassRange
    {
-      public ClassRange(int pos, String className, String text)
+      public ClassRange(int pos, String className, String text, Hyperlink hyperlink)
       {
          clazz  = className;
          start = pos;
          length = text.length();
+         hyperlink_ = hyperlink;
          element = Document.get().createSpanElement();
+         if (hyperlink_ != null) 
+         {
+            Event.sinkEvents(element, Event.ONCLICK);
+            Event.setEventListener(element, event -> {
+               server_.consoleFollowHyperlink(hyperlink_.url_, text, hyperlink_.params_, new VoidServerRequestCallback());
+            });
+
+            // for now, perhaps a dedicated style would be better
```

Comment:
I'm not sure where to add a new specific style for links ? cc @gtritchie

Reply:
The console styles (along with the terminal, aka xterm) are part of the themes, so I think you'd need to add a new style to all the various themes you see in Options / Appearances, and also deal with what to do with third-party themes that don't have the new one. For now, if it looks ok, I think using the existing xterm style is reasonable. @MariaSemple did a lot of the theming work so she might have thoughts (and know if I'm totally off base here).

Reply:
I'll look into adding a `xtermHyperlink` style. We might want to fine tune how they are displayed. For example `cli` (not using `style_hyperlink()` yet, until https://github.com/r-lib/cli/issues/386) uses: 

![image](https://user-images.githubusercontent.com/2625526/149095456-2ce22cb8-2231-4ef6-b5d6-12fc2a1f03f4.png)

Reply:
Gary is on base :). We should have a fallback/default in case user-themes are missing a rule for this class, but we should add a rule for each of the themes we ship (see here: https://github.com/rstudio/rstudio/blob/main/src/cpp/session/resources/themes/)